ksqldb-engine/src/main/java/io/confluent/ksql/engine/rewrite/DataSourceExtractor.java

```diff
@@ -126,50 +94,34 @@ public Void visitRelation(final Relation relation, final Void ctx) {
 
     @Override
     public Void visitAliasedRelation(final AliasedRelation relation, final Void ctx) {
-      fromAlias = relation.getAlias();
-      fromName = ((Table) relation.getRelation()).getName();
+      final SourceName fromName = ((Table) relation.getRelation()).getName();
       final DataSource source = metaStore.getSource(fromName);
       if (source == null) {
         throw new KsqlException(fromName.text() + " does not exist.");
       }
 
-      allSources.add(source);
+      allSources.add(new AliasedDataSource(relation.getAlias(), source));
```

Comment:
I assume this is in place of the join code, which you've removed.  Why was join special-cased before?

Reply:
you are right, this does the same thing, but is just cleaner IMO - when there were just two sides of a join it wasn't too bad to special case it, but I don't think there was any good reason to